### PR TITLE
PRODENG-1577: Kube deployment

### PR DESCRIPTION
**Pull request by raynorg**
## Screenshots

|  Before  |  After  |
|--------|--------|
|||

## Testing in `kajabi-products`

1. (**LOW/MEDIUM/HIGH/BREAKING**) Description of the change and its impact with QA as the audience.
   - [ ] One more examples of the component in use to either test the change or verify the change has not had adverse effects.

### CODE REVIEW

**Comment by ju-Skinner** [comment]
@raynorg I see you managed to get a Green build...How are we looking in terms of getting this merged?

**Comment by ju-Skinner** [comment]
Just a heads up I have merged #1641, which updates Ruby to v2.7.7 and Rails to 6.0.6

File: packages/sage-assets/Dockerfile [hunk 1]
```diff
@@ -3,7 +3,7 @@ FROM ruby:3.1.2-alpine
 ARG RAILS_ENV="production"
 ARG NONROOT_UID="1000"
 ARG NONROOT_GID="1000"
-ARG NODE_VERSION="15.14.0"
+ARG NODE_VERSION="14.21.3"
```
Comment (rwc9u) [inline packages/sage-assets/Dockerfile:6]: why go backwards?

Comment (ju-Skinner) [reply packages/sage-assets/Dockerfile:6]: Good catch @rwc9u

File: docs/config/application.rb [hunk 2]
```diff
@@ -33,9 +33,10 @@ class Application < Rails::Application
     # config.active_record.raise_in_transactional_callbacks = true
 
     # Storybook deploy url to differentiate in different environments
-    config.storybook_root_url = "https://sage-lib-storybook.herokuapp.com/?path=/docs/"
+    config.storybook_root_url = "https://sage-lib-storybook.production.kajabi.farm/?path=/docs/"
 
     # Sassdocs deploy url to differentiate in different environments
-    config.sassdocs_root_url = "https://sage-lib-sassdocs.herokuapp.com/"
+    config.sassdocs_root_url = "https://sage-lib-sassdocs.production.kajabi.farm/"
```
Comment (prsimp) [inline docs/config/application.rb:39]: Why are these being set here vs. in the `config/environments/production.rb` file?

**Review by prsimp — APPROVED** [review]
Looks good to me. A few quick questions, but I'm good with no action responses, just curious:

* Why set the "default" value to production values vs. using the environmental config?
* If we're dropping the qa/dev environments, why not remove those environmental config files?

Nice work getting this over the line @voodooGQ @raynorg @ju-Skinner 👍

File: docs/config/application.rb (hunk 2 above)
Comment (ju-Skinner) [reply docs/config/application.rb:39]: @prsimp 
I am not disagreeing with you, but the current implementation reads from the config files.

https://github.com/Kajabi/sage-lib/blob/maindocs/app/views/application/_home_code.html.erb#L14

https://github.com/Kajabi/sage-lib/blob/main/docs/app/views/application/_footer.html.erb#L22

https://github.com/Kajabi/sage-lib/blob/main/docs/app/helpers/application_helper.rb#L38

File: .github/workflows/release-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,134 @@
+name: Release-Deploy
+on:
+  # !!Emergency override!! uncomment below and access the actions tab in GitHub to run a release/deploy
+  # workflow_dispatch:
+  push:
+    branches:
+      - main
+      - develop
+      - PRODENG-1577/kube_deployment
+jobs:
```
Comment (ju-Skinner) [inline .github/workflows/release-deploy.yml:10]: Is there a reason `lint,test, and build` were removed from this workflow?

File: .github/workflows/release-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  release_deploy:
```
Comment (ju-Skinner) [inline .github/workflows/release-deploy.yml:11]: Having `release_deploy` as one big job will require rerunning the whole process should one step fail.

File: docs/Dockerfile [hunk 5]
```diff
@@ -8,6 +8,10 @@ ARG NODE_VERSION="16.20.2"
 ARG YARN_VERSION="1.22.18"
 ARG ARCH="x64"
 ARG GITHUB_TOKEN
+ARG SECRET_KEY_BASE="foobarbaz"
```
Comment (ju-Skinner) [inline docs/Dockerfile:11]: Is this the value we want to use?

File: .github/workflows/release-deploy-v2.yml
Comment (ju-Skinner) [inline .github/workflows/release-deploy-v2.yml:1]: 📓  This was used for the current heroku deployments.

File: docs/config/application.rb (hunk 2 above)
Comment (voodooGQ) [reply docs/config/application.rb:39]: Soooo..... leave as is? Or change?

File: .github/workflows/release-deploy-v2.yml
Comment (voodooGQ) [reply .github/workflows/release-deploy-v2.yml:1]: K. Won't be necessary once we merge and verify, so think it's safe to leave out. But you let me know.

File: .github/workflows/release-deploy.yml (hunk 3 above)
Comment (voodooGQ) [reply .github/workflows/release-deploy.yml:10]: This is a net-new file, so not sure how it was "removed". Do you want me to add the job as a dep?

File: .github/workflows/release-deploy.yml (hunk 4 above)
Comment (voodooGQ) [reply .github/workflows/release-deploy.yml:11]: What would you like here? I have no stake in this game, so need some direction on how you'd like it handled. Otherwise feel free to make changes and commit to this PR if you have a game plan.

File: docs/Dockerfile (hunk 5 above)
Comment (voodooGQ) [reply docs/Dockerfile:11]: 🤷 it's what was in there. Would you like something different?

File: docs/config/application.rb (hunk 2 above)
Comment (voodooGQ) [reply docs/config/application.rb:39]: @prsimp dev !== development. There was some finagling to get it to work right on Argo, but now that we're not using `dev` as an env the `dev` references have been removed. `development` is still in use for local.

@ju-Skinner mentioned that QA was in use in some fashion, but it's not a deployed environment. Happy to remove the `rb` file too if that's what we want to do, but would need an approval from @ju-Skinner

**Comment by voodooGQ** [comment]
Forgot we merged the Node 18 PR into develop, so gotta go through and update the Dockerfiles. BRB...

File: docs/Dockerfile (hunk 5 above)
Comment (ju-Skinner) [reply docs/Dockerfile:11]: 🤷🏾 just called it out. I don't know either. lol

File: .github/workflows/release-deploy.yml (hunk 3 above)
Comment (ju-Skinner) [reply .github/workflows/release-deploy.yml:10]: Sorry, I wasn't clear. In comparison to `release-deploy-v2`, we no longer run those in this workflow. Is there a reason it wasn't included?

**Comment by voodooGQ** [comment]
Nevermind, can't really do that until the Node 18 branch is in `main`. Will have to address once it's been launched.

File: .github/workflows/release-deploy.yml (hunk 3 above)
Comment (voodooGQ) [reply .github/workflows/release-deploy.yml:10]: Mainly because it wasn't in there when I took over. I didn't really go through all the details because we had some environments working, my goal was simply to get the rest of them working. So, no, no particular reason it wasn't included. Happy to add it if that's what you want, just give me the word.

Comment (voodooGQ) [reply .github/workflows/release-deploy.yml:10]: Going through the file diff here I don't see `lint/test/build` being referenced in any of the old ones. I do see [this](https://github.com/Kajabi/sage-lib/pull/1687/files#diff-b4933e98d656990070dfbba80c7921824d127f73e4218be90be75a9cbad9b05eL160) but that's referring to the `lint`, `test`, and `build` steps in that specific workflow.

Not saying lint/test/build would be a bad thing by any means, but just wondering if it's actually necessary? Your `develop` and `main` are protected right (Could be wrong about that, legitimately asking)? And `lint/test/build` fires on PR, so it would have ran before it got merged _technically_. Doesn't cover all edge cases but it should work as expected with that kind of setup right?

Either way I'm fine adding it, couple line change and adds some extra protection. But not sure if it's strictly _necessary_

**Review by rwc9u — APPROVED** [review]

File: docs/Dockerfile (hunk 5 above)
Comment (rwc9u) [reply docs/Dockerfile:11]: You normally need some value in that environment variable when running any asset precompile or rails command. It's set via ENV var in most apps in a production environment, but you just have to have something. It could be changed to `"setting_random_secret_for_asset_precompile"` if you want

File: .github/workflows/release-deploy.yml (hunk 3 above)
Comment (voodooGQ) [reply .github/workflows/release-deploy.yml:10]: @ju-Skinner ^

File: .github/workflows/release-deploy.yml (hunk 4 above)
Comment (voodooGQ) [reply .github/workflows/release-deploy.yml:11]: @ju-Skinner ^

Comment (ju-Skinner) [reply .github/workflows/release-deploy.yml:11]: Could it resemble what was in `release-deploy-v2`?

Comment (voodooGQ) [reply .github/workflows/release-deploy.yml]: Added

**Comment by voodooGQ** [comment]
This is mergable at this point, waiting for some information on the Kubernetes side before we merge that one, and then we'll merge this one. Doing it in this order so we don't drop the Heroku stuff before the K8s is ready.
